### PR TITLE
add Post and Comment

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

**Comment by MixailE** [comment]
[Add data caching to notifications backend](https://trello.com/c/GV7oBveW/2513-add-data-caching-to-notifications-backend)

File: src/controllers/notifications.ts [hunk 1]
```diff
@@ -42,30 +38,38 @@ export default class NotificationsController {
...
+      },
+      defaultProjection
     )
 
     return {
       currentBlock,
-      modifiedCommentSinceLastCheck,
-      modifiedPostsSinceLastCheck,
+      modifiedCommentSinceLastCheck: modifiedCommentSinceLastCheck.map(
+        (comment) => comment.toObject({ virtuals: true })
```
Comment (smyzin) [inline src/controllers/notifications.ts:65]: The function can be moved to a separate method, imo

File: src/helpers/parse.ts [hunk 2]
```diff
@@ -1,48 +1,58 @@
...
+    .map(Number)
+
   return {
-    commentId,
-    feedId,
     metadata: structToCid(metadata),
-    numberOfComments: numberOfComments.toNumber(),
-    postId,
-    replyTo: replyTo.toNumber(),
+    numberOfComments: numberOfComments,
```
Comment (smyzin) [inline src/helpers/parse.ts:16]:
```
return {
  metadata: structToCid(metadata),
  numberOfComments,
  replyTo,
  sender,
  threadId,
  timestamp,
}
```

File: src/helpers/saveBlockchainEvents.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    await PostModel.findOneAndUpdate(
+      { transactionHash: post.transactionHash },
+      post,
+      {
+        new: true,
+        upsert: true,
+      }
+    )
+  } catch (e) {
+    console.error(e)
```
Comment (smyzin) [inline src/helpers/saveBlockchainEvents.ts:23]: Do we have Sentry to handle it in this project?

File: src/models/Post.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  feedId: 1,
+  metadata: 1,
+  numberOfComments: 1,
+  postId: 1,
+  replyTo: 1,
+  sender: 1,
+  threadId: 1,
+  timestamp: 1,
+  username: 1,
+}
```
Comment (smyzin) [inline src/models/Post.ts:49]: Is it a good place to store this value? Maybe, let's create a separate file for this

**Review by smyzin — APPROVED** [review]
Let's clean up, and we are good to go

File: src/helpers/parse.ts [hunk 5]
```diff
@@ -1,48 +1,58 @@
...
-  const feedId = args[0].toNumber()
-  const postId = args[1].toNumber()
-  const commentId = args[2].toNumber()
-  const comment = args[3]
-  const [sender, metadata, timestamp, threadId, replyTo, numberOfComments] =
-    comment
+function transform(post: PostStructOutput) {
+  const [sender, metadata] = post
+  const [timestamp, threadId, replyTo, numberOfComments] = post
+    .slice(2)
```
Comment (T-Damer) [inline src/helpers/parse.ts:11]: Let's not slice, simply ignore when destructuring

File: src/models/Post.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  public numberOfComments!: number
+
+  public get username() {
+    return generateRandomName(this.sender)
+  }
+}
+
+export const PostModel = getModelForClass(Post)
+
+export const defaultProjection = {
```
Comment (T-Damer) [inline src/models/Post.ts:37]: `defaultPostProjection`

File: src/models/Post.ts (hunk 4 above)
Comment (T-Damer) [reply src/models/Post.ts:49]: I think it's fine

**Pull request merged**